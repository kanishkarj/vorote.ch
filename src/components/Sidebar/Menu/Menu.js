// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';
import { marginBottom } from '../../../utils/get-margin';

const StyledMenu = styled.nav`
  ${marginBottom(1)}
`;

const MenuList = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
`;

const MenuListItem = styled.li`
  padding: 0;
  margin: 10px 0;
`;

const ListItemLink = styled(Link)`
  font-size: ${(props) => props.theme.fontSizes.base}px;
  color: var(--color-text);
  font-weight: normal;
  border: 0;

  &:hover,
  &:focus {
    color: var(--color-primary);
    border-bottom: 1px solid var(--color-primary);
  }

  &:.active {
    color: var(--color-text);
    border-bottom: 1px solid var(--color-text);
  }
`;


type Props = {
  menu: {
    label: string,
    path: string
  }[]
};

const Menu = ({ menu }: Props) => (
  <StyledMenu>
    <MenuList>
      {menu.map((item) => (
        <MenuListItem key={item.path}>
          <ListItemLink to={item.path} activeClassName='active'>
            {item.label}
          </ListItemLink>
        </MenuListItem>
      ))}
    </MenuList>
  </StyledMenu>
);

export default Menu;
