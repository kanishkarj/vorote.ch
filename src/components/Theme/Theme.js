import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
