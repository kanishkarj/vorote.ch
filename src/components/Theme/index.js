// @flow strict
export { default } from './Theme';
