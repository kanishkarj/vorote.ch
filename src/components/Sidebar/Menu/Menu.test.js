// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Menu from './Menu';
import 'jest-styled-components';
import Theme from '../../Theme';

describe('Menu', () => {
  const props = {
    menu: [
      {
        label: 'Item 0',
        path: '/#0/'
      },
      {
        label: 'Item 1',
        path: '/#1/'
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer.create(
    <Theme>
      <Menu {...props} />
    </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
