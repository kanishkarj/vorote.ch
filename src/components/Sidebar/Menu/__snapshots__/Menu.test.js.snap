// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
.c0 {
  margin-bottom: 26px;
}

.c1 {
  list-style: none;
  padding: 0;
  margin: 0;
}

.c2 {
  padding: 0;
  margin: 10px 0;
}

.c3 {
  font-size: 16px;
  color: var(--color-text);
  font-weight: normal;
  border: 0;
}

.c3:hover,
.c3:focus {
  color: var(--color-primary);
  border-bottom: 1px solid var(--color-primary);
}

.c3:.active {
  color: var(--color-text);
  border-bottom: 1px solid var(--color-text);
}

<nav
  className="c0"
>
  <ul
    className="c1"
  >
    <li
      className="c2"
    >
      <a
        className="c3"
        href="/#0/"
      >
        Item 0
      </a>
    </li>
    <li
      className="c2"
    >
      <a
        className="c3"
        href="/#1/"
      >
        Item 1
      </a>
    </li>
  </ul>
</nav>
`;
